### PR TITLE
Use ldexp from cmath instead

### PR DESCRIPTION
**Pull request by christianpoveda**
Closes https://github.com/rust-lang/miri/issues/821
r? @RalfJung

### CODE REVIEW

File: src/shims/foreign_items.rs [hunk 1]
```diff
@@ -594,10 +594,17 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 // FIXME: Using host floats.
                 let x = f64::from_bits(this.read_scalar(args[0])?.to_u64()?);
                 let exp = this.read_scalar(args[1])?.to_i32()?;
-                // FIXME: We should use cmath if there are any imprecisions.
-                let n = x * 2.0f64.powi(exp);
+                let n = x.ldexp(exp);
```
Comment (RalfJung) [inline src/shims/foreign_items.rs]: Where are those methods coming from now...?

Comment (christianpoveda) [reply src/shims/foreign_items.rs]: https://github.com/rust-lang/rust/pull/63386/files#diff-f60977ab00fd9ea9ba7ac918e12a8f42R613

Comment (RalfJung) [reply src/shims/foreign_items.rs]: Oh... that's not at all what I suggested or had in mind though.^^

I said you should use the `extern` functions. Like, access it the same way the libcore test suite does.

Comment (RalfJung) [reply src/shims/foreign_items.rs]: I literally said "import the function via `extern` instead". No idea how you construed that into adding new APIs to libstd...

Comment (christianpoveda) [reply src/shims/foreign_items.rs]: It is a long boring story

**Comment by RalfJung** [comment]
Yes that looks right, thanks. :)

@bors r+

**Pull request merged**